test/testactivitylistmodel.cpp

```diff
@@ -452,164 +477,166 @@ private slots:
+        testSyncResultErrorActivity._message = QStringLiteral("/path/to/thingy");
+        testSyncResultErrorActivity._link = QStringLiteral("/path/to/thingy");
+        testSyncResultErrorActivity._accName = accountState->account()->displayName();
+
+        testSyncFileItemActivity._id = 3;
+        testSyncFileItemActivity._type = OCC::Activity::SyncFileItemType; //client activity
+        testSyncFileItemActivity._status = OCC::SyncFileItem::Success;
+        testSyncFileItemActivity._dateTime = QDateTime::currentDateTime();
+        testSyncFileItemActivity._message = QStringLiteral("Sample file successfully synced text");
+        testSyncFileItemActivity._link = accountState->account()->url();
+        testSyncFileItemActivity._accName = accountState->account()->displayName();
+        testSyncFileItemActivity._file = QStringLiteral("xyz.pdf");
+
+        testFileIgnoredActivity._id = 4;
+        testFileIgnoredActivity._type = OCC::Activity::SyncFileItemType;
+        testFileIgnoredActivity._status = OCC::SyncFileItem::FileIgnored;
+        testFileIgnoredActivity._dateTime = QDateTime::currentDateTime();
+        testFileIgnoredActivity._subject = QStringLiteral("Sample ignored file sync text");
+        testFileIgnoredActivity._link = accountState->account()->url();
+        testFileIgnoredActivity._accName = accountState->account()->displayName();
+        testFileIgnoredActivity._folder = QStringLiteral("thingy");
+        testFileIgnoredActivity._file = QStringLiteral("test.txt");
     };
 
     // Test receiving activity from server
     void testFetchingRemoteActivity() {
-        TestingALM model;
-        model.setAccountState(accountState.data());
-        QAbstractItemModelTester modelTester(&model);
+        const auto model = testingALM();
+        QCOMPARE(model->rowCount(), 0);
 
-        QCOMPARE(model.rowCount(), 0);
-
-        model.setCurrentItem(FakeRemoteActivityStorage::instance()->startingIdLast());
-        model.startFetchJob();
-        QSignalSpy activitiesJob(&model, &TestingALM::activitiesProcessed);
+        model->setCurrentItem(FakeRemoteActivityStorage::instance()->startingIdLast());
+        model->startFetchJob();
+        QSignalSpy activitiesJob(model.data(), &TestingALM::activitiesProcessed);
         QVERIFY(activitiesJob.wait(3000));
-        QCOMPARE(model.rowCount(), 50);
+        QCOMPARE(model->rowCount(), 50);
     };
 
     // Test receiving activity from local user action
     void testLocalSyncFileAction() {
-        TestingALM model;
-        model.setAccountState(accountState.data());
-        QAbstractItemModelTester modelTester(&model);
-
-        QCOMPARE(model.rowCount(), 0);
-
-        OCC::Activity activity;
-
-        model.addSyncFileItemToActivityList(activity);
-        QCOMPARE(model.rowCount(), 1);
-
-        const auto index = model.index(0, 0);
-        QVERIFY(index.isValid());
+        testActivityAdd(&TestingALM::addSyncFileItemToActivityList, testSyncFileItemActivity);
```

Comment:
this test and the few next ones would be better written as data driven tests with different data instead of doing an utility method and calling it manually several time
see https://doc.qt.io/qt-5/qttestlib-tutorial2-example.html

Reply:
I am not sure, what we want to test here is that the activity is added to the activity list correctly, and that it returns the right data, right? For the latter we have the `data` testing function, in the test slots that take these methods we are testing if the activity is actually added properly

I am not sure what benefit using the QTest data table would bring, in our case it seems it would add more complexity as we would need to retain the activities anyway, add the activity data we want to test to the QTest table, and then test the table rows vs. just testing the data we get from the table with the data in the activity we are retaining directly

Reply:
I would say that the data driven tests are the thing you have to use when you want to run teh same code multiple time with different input values
so sounded to me like the solution here

Reply:
I don't see a lot of reason for the table as we need to keep the activity objects around and these are never changed. We can just compare these directly, at least as I see it -- we instantiate the activities once in the constructor and that is it, we don't need to re-create new activities in each test

The polymorphic way of testing the methods is just there to remove the need for repeating the same lines of code checking for things over and over